wurst/lib/ItemRecipe.wurst

```diff
@@ -0,0 +1,527 @@
+    ItemRecipe prevRecipe = eventRecipe
+    unit prevUnit = eventUnit
+    item prevItem = eventItem
+    let prevRecipeData = eventRecipeData
+
+    eventRecipe = recipe
+    eventUnit = u
+    eventItem = itm
+    eventRecipeData = recipeData
+
+    evt.evaluate()
+
+    eventRecipe = prevRecipe
+    eventUnit = prevUnit
+    eventItem = prevItem
+    eventRecipeData = prevRecipeData
+
+/** Stores information about item required for recipe assembly. */
+public class ItemData
+    int itemTypeId
+    boolean remove
+    int charges
+    int index
+
+    construct(int itemTypeId, boolean remove, int charges, int index)
+        this.itemTypeId = itemTypeId
+        this.remove = remove
+        this.charges = charges
+        this.index = index
+    
+    construct(thistype base)
+        this.itemTypeId = base.itemTypeId
+        this.remove = base.remove
+        this.charges = base.charges
+        this.index = base.index
+
+/** Item recipe, collection of items that can be combined for powerful rewards. */
+public class ItemRecipe
+    private LinkedList<ItemData> ingredients
+    private int reward
+    int charges
+    boolean ordered
+    boolean permanent
+    boolean pickUp
+    UnitRequirement requirement
+    private int abilityId
+    private int count
+    private boolean batch
+
+    construct(int reward, int charges, boolean ordered, boolean permanent, boolean pickUp)
+        ingredients = new LinkedList<ItemData>()
+        this.reward = reward
+        this.charges = charges
+        this.ordered = ordered
+        this.permanent = permanent
+        this.pickUp = pickUp
+        requirement = null
+        abilityId = 0
+
+        resultToRecipe.saveInt(reward, this castTo int)
+
+    ondestroy
+        LinkedList<ItemRecipe> recipes
+        int itemTypeId
+
+        setAbility(0)
+        resultToRecipe.removeInt(reward)
+
+        for data in ingredients
+            itemTypeId = data.itemTypeId
+            recipes = recipeMap.get(itemTypeId)
+            recipes.remove(this)
+            if recipes.isEmpty()
+                destroy recipes
+                recipeMap.remove(itemTypeId)
+
+        destroy ingredients
+
+    /** Number of items required for the recipe. */
+    function count() returns int
+        return count
+
+    /** Whether specified item type is a part of recipe. */
+    function inList(int itemTypeId) returns boolean
+        for data in ingredients
+            if data.itemTypeId == itemTypeId
+                return true
+        return false
+
+    /** Returns shallow copy of item recipe data. */
+    function getIngredients() returns LinkedList<ItemData>
+        return new LinkedList<ItemData>(ingredients)
+
+    /** Returns item type of this recipe reward. */
+    function getResult() returns int
+        return reward
+
+    /** Starts single-reference counted batch. Allows to assign multiple items to the same item slot. */
+    function startBatch()
+        if not batch
```

Comment:
Currently this check doesn't do anything. Either way, `batch` is `true` afterwards so simply assigning it is sufficient. If you plan on adding an `else` branch, such as for error logging, please add a `TODO` comment for that.

Reply:
Precisely, in vjass version I have debug messages added. This is still not included in wurst version.